source/linux/Foundational_Components/U-Boot/UG-Memory.rst

```diff
@@ -1,1052 +1,1216 @@
+           created partition.
+         - Set the boot mode switches to USB host boot mode (Refer to  **Initialization**
+           chapter of TRM for boot switch details)
+         - Make sure USB0 port in DRP mode: SW2[2:3] = 00
+         - Connect the USB Mass storage device with the bootloader images and boot up
+           the board.
+
+   .. note::
+
+      While using :command:`usb reset` or :command:`usb start` commands in U-Boot, or
+      booting from a USB Mass storage device, some of the USB devices fail to get
+      detected. This issue is seen because these USB devices are failing to follow the
+      spec for power good delay. It can be resolved by overriding the **power good delay**
+      using the environment variable **usb_pgood_delay**, setting it to 2000 should be
+      good enough for all cases.
+
+Boot Linux from USB storage
+==============================
+
+.. ifconfig:: CONFIG_part_family not in ('General_family', 'AM335X_family', 'AM437X_family', 'AM57X_family', 'J7_family', 'AM62X_family', 'AM64X_family')
+
+   Booting Linux from USB storage is not supported for |__PART_FAMILY_DEVICE_NAMES__|
+   or documentation is pending, please reach out to:  `Help e2e <https://e2e.ti.com//>`__
+   for additional information.
+
+.. ifconfig:: CONFIG_part_family in ('General_family', 'AM335X_family', 'AM437X_family', 'AM57X_family')
+
+   To load the Linux Kernel and rootfs from USB rather than SD/MMC card on
+   AMx/DRA7x EVMs, if we assume that the USB device is partitioned the same
+   way as an SD/MMC card is, we can utilize the :command:`mmcboot` command
+   to boot. To do this, perform the following steps:
+
+   .. code-block:: console
+
+      => usb start
+      => setenv mmcroot /dev/sda2 ro
+      => run mmcargs
+      => run bootcmd_usb
+
+   The USB drivers in Linux kernel need to be built-in (default modules). The
+   configuration changes are:
+
+   .. code-block:: Kconfig
+
+      CONFIG_USB=y
+      CONFIG_USB_XHCI_HCD=y
+      CONFIG_USB_XHCI_PCI=y
+      CONFIG_USB_XHCI_PLATFORM=y
+      CONFIG_USB_STORAGE=y
+      CONFIG_USB_DWC3=y
+      CONFIG_USB_DWC3_HOST=y
+      CONFIG_USB_DWC3_KEYSTONE=y
+      CONFIG_EXTCON=y
+      CONFIG_EXTCON_USB_GPIO=y
+      CONFIG_SCSI_MOD=y
+      CONFIG_SCSI=y
+      CONFIG_BLK_DEV_SD=y
+
+   The USB should have boot partition of FAT32 format, and rootfs partition
+   of EXT4 format. The boot partition must contain the following images:
+
+   - keystone-<platform>-evm.dtb
+   - skern-<platform>.bin
+   - k2-fw-initrd.cpio.gz
+   - zImage
+
+   The rootfs partition contains the filesystem from ProcSDK release package. Extracting
+   failes to the rootfs parititon should be done as a root user.
+
+   .. code-block:: console
+
+      $ mkdir /mnt/temp
+      $ mount -t ext4 /dev/sdb2 /mnt/temp
+      $ cd /mnt/temp
+      $ tar xvf <Linux_Proc_Sdk_Install_DIR>/filesyste/tisdk-server-rootfs-image-k2hk-evm.tar.xz
```

Comment:
https://github.com/TexasInstruments/processor-sdk-doc/pull/114#discussion_r1884599522

Reply:
Also, isn't this missing the `J` switch to actually filter the archive through `xz` ?

Reply:
Nope, not necessary. AFAIK GNU tar should identify the format on its own, no need for the J flag. This command listed here should work.